src/FSharp.Data.GraphQL.Client/GraphQLClientConnection.fs

```diff
@@ -8,16 +8,16 @@ open System.Net.Http
 
 /// The connection component for GraphQL client operations.
 [<AllowNullLiteral>]
-type GraphQLClientConnection(client : HttpClient, ownsClient : bool) =
-    /// Creates a new instance of GraphQLClientConnection, using a provided HttpClient to be wrapped inside.
-    /// The provided HttpClient is not owned by the connection, so it must be disposed when it is not necessary anymore.
-    new(client) = new GraphQLClientConnection(client, false)
+type GraphQLClientConnection(invoker : HttpMessageInvoker, ownsInvoker : bool) =
+    /// Creates a new instance of GraphQLClientConnection, using a provided System.Net.Http.HttpMessageInvoker to be wrapped inside.
+    /// The provided System.Net.Http.HttpMessageInvoker is not owned by the connection, so it must be disposed when it is not necessary anymore.
+    new(invoker) = new GraphQLClientConnection(invoker, false)
```

Comment:
Could you also add a constructor that accept `HttpClient` so that library user does not need to upcast

Reply:
Sure.